# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
# Commit: d07def1365a4
# Full Hash: d07def1365a49dc8ecbf5b6c10fdfbd5ec750932
# Author: pstanciu <pstanciu@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing WPT failures @lfedcm-identity.create-store-collect.tentative.sub.https.html
#   
#   This reverts commit 03d4474e360f28be0fb075379f4e4535bd34b252.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4e25ddaadc86 -r d07def1365a4 toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
--- a/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Wed Jun 11 13:42:13 2025 +0000
+++ b/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Wed Jun 11 18:04:27 2025 +0300
@@ -31,6 +31,22 @@
     // Delete the entry for all tuples (rpPrincipal, idpPrincipal, *).
     void disconnect(in nsIPrincipal rpPrincipal, in nsIPrincipal idpPrincipal);
 
+     // Store a "lightweight" identity credential
+     [noscript]
+    void storeIdentityCredential(in IPCIdentityCredentialRef credential);
+
+    // Retrieve all "lightweight" identity credentials stored by these IDPs
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentials(in Array<nsIPrincipal> idpPrincipal);
+
+    // Retrieve all "lightweight" identity credentials stored with this type, for the provided RP
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentialsOfType(in ACString type);
+
+    // Delete a  "lightweight" identity credential
+     [noscript]
+    void deleteIdentityCredential(in IPCIdentityCredentialRef credential);
+
     // Delete all data in this service.
     void clear();
 